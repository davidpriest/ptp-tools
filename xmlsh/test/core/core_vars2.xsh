# core_vars.xsh
# test variables

if [ -z "$TMPDIR" -o ! -d "$TMPDIR" ] ; then
	echo Temp directory required TMPDIR: $TMPDIR
	exit 1;
fi





# turn off indentation for this test
set +indent


# clear just in case they were exported
unset A AA
unset I

# Test expended [] notation
A=(a b c)
I=2

echo "A[2] is" ${A[$I]} 

# Test empty variables expand to nothing
A=
B=test
C=

echo $A $B $C
set $A $B $C
echo $#

# Test we cannot set to the empty variable name using >{}
echo SNH >{""}
blank=""
echo SNH >{$blank}
exit 0
