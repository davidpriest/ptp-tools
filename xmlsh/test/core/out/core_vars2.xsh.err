[C:/Work/DEI/xmlsh/trunk/test/core/core_vars2.xsh line: 37]
Exception running: echo
org.xmlsh.core.CoreException: Invalid blank name for output variable
[C:/Work/DEI/xmlsh/trunk/test/core/core_vars2.xsh line: 39]
Exception running: echo
org.xmlsh.core.CoreException: Invalid blank name for output variable
