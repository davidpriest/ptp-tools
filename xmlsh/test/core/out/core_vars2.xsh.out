A[2] is b
test
1
